js/options.js

```diff
@@ -35,14 +35,36 @@
                         view.confirmNumber.bind(view),
                         view.incrementCounter.bind(view)
                     ).then(function() {
+
+                        // helper function to make time-outing promises that
+                        // wait for event and then "clean" after themselves
+                        function timeoutListenerPromise(event) {
+                            return new Promise(function(resolve) {
+                                var timeout;
+                                var listener = function() {
+                                    clearTimeout(timeout);
+                                    resolve();
+                                }
+                                timeout = setTimeout(function() {
+                                    bg.removeEventListener(event, listener);
+                                    resolve();
+                                }, 60000);
+                                bg.addEventListener(event, listener);
+                            });
+                        }
+
```

Comment:
Hm. It's actually wrong now, the listener should be removed not in timeout, but after resolve. 

I will look if it won't be better and cleaner to add functionality similar to EventEmitter's "once", that would be easier than all this craziness